deployer/orb.yaml

```diff
@@ -1,6 +1,85 @@
+      exactly with those used by pip.
+    parameters:
+      verify_version:
+        default: CIRCLE_TAG
+        description: >
+          The string or string-name of env var which holds the version string
+          to parse with pip Version
+        type: string
+      should_upgrade_pip:
+        type: boolean
+        default: false
+    steps:
+      - when:
+          condition: <<parameters.should_upgrade_pip>>
+          steps:
+            - run: python3 -m pip install --upgrade pip
+      - run:
+          name: Validate tag format for pip
+          shell: /usr/local/bin/python
+          command: |
+            import os
+            from pip._vendor.packaging.version import Version
+            try:
+              # See if verify_version is env var name holding version value
+              tag = os.environ['<<parameters.verify_version>>']
+            except KeyError:
+              # Check if verify_version itself was a version string
+              tag = '<<parameters.verify_version>>'
+            v = Version(tag)
+            assert tag == str(v), f'Original tag ({tag}) should have been {str(v)}'
+            assert v.is_devrelease or not v.local, f'tag should not contain \'+{v.local}\' unless dev version'
+
+  verify-tag-matches-deploy:
+    parameters:
+      folder:
+        default: '.'
+        description: >
+          Path to the folder containing the Python package.
+        type: string
+      verify_version:
+        default: CIRCLE_TAG
+        description: >
+          If set to a non-empty value, fails the deployment if the version
+          being deployed does not match. Can be used as a last-step sanity
+          check.
+        type: string
+    steps:
+      - run:
+          name: verify tag builds match deploy version
+          command: |
+            if [[ ! -z "${<<parameters.verify_version>>}" ]]; then
+                code_version=$(find "<<parameters.folder>>/dist" -iname '*.tar.gz' | awk -F- '{sub(/\.tar\.gz/, ""); print $NF}')
+                code_version_alt="${code_version/.dev/-dev.}"
+                if [[ <<parameters.verify_version>> != ${code_version} &&
+                      <<parameters.verify_version>> != ${code_version_alt} ]]; then
+                    echo "Version ${code_version} does not match tag ${<<parameters.verify_version>>}"
+                    exit 1
+                fi
+            fi
+
+  upload:
+    parameters:
+      folder:
+        default: '.'
+        description: >
+          Path to the folder containing the Python package.
+        type: string
+      gemfury_token:
+        default: GEMFURY_TOKEN
+        description: >
+          Name of the env var containing the Gemfury token.
+        type: env_var_name
+    steps:
+      - run: ls "<<parameters.folder>>/dist" | xargs -I{} curl -F package=@<<parameters.folder>>/dist/{} https://${<<parameters.gemfury_token>>}@push.fury.io/talkiq/
```

Comment:
This reminds me that we should actually make a ticket for addressing silent failures when we fail to upload to gemfury. For instance, we might silently fail with a message like this:
```
#!/bin/bash -eo pipefail
ls "core/dist" | xargs -I{} curl -F package=@core/dist/{} https://${GEMFURY_TOKEN}@push.fury.io/talkiq/
~~> Processing package upload
    voiceai-formatter-core-6.1.0.dev10.tar.gz ... ok
~~> Processing package upload
    voiceai_formatter_core-6.1.0.dev10-py2.py3-none-any.whl ... failed
CircleCI received exit code 0
```

Reply:
https://switchcomm.atlassian.net/browse/VAE-2276